lib/puppet/provider/service/systemd.rb

```diff
@@ -7,6 +7,7 @@
 
   defaultfor :osfamily => [:archlinux]
   defaultfor :osfamily => :redhat, :operatingsystemmajrelease => "7"
+  defaultfor :osfamily => :redhat, :operatingsystem => :fedora, :operatingsystemmajrelease => ["17", "18", "19", "20", "21"]
```

Comment:
I'm worried we are going to forget to increment these as new versions are released, especially because fedora releases are more frequent than rhel.

One way may be to confine the acceptance tests to `/el\-|centos\-|fedora\-/` and blacklist older versions, and fail versions that we don't support yet:

```
agents.each do |agent|
  case on(agent, facter('osfamily')).stdout.chomp
  when 'fedora'
    majrelease = on(agent, facter('operatingssystemmajrelease')).stdout.chomp.to_i
    if majrelease < 17
      skip_test "Test not applicable to fedora #{majrelease}"
    elsif majrelease > 21
      fail_test "Provider does not support fedora #{majrelease}"
    end
  when 'redhat'
   ...
```

Actually now that I write this, I think we could do this check in specs to handle the fedora 22, etc cases.